packages/dds/tree/src/test/shared-tree/opSize.bench.ts

```diff
@@ -18,50 +18,36 @@ import {
 
-import {
-	AllowedUpdateType,
-	type FieldKey,
-	type JsonableTree,
-	type Value,
-	forEachNode,
-	moveToDetachedField,
-	rootFieldKey,
-} from "../../core/index.js";
-import { SchemaBuilder, leaf } from "../../domains/index.js";
+import type { FieldKey, Value } from "../../core/index.js";
 import { typeboxValidator } from "../../external-utilities/index.js";
-import {
-	TreeCompressionStrategy,
-	cursorForJsonableTreeNode,
-} from "../../feature-libraries/index.js";
-import type { ISharedTree, ITreeCheckout, SharedTree } from "../../shared-tree/index.js";
+import { TreeCompressionStrategy } from "../../feature-libraries/index.js";
+import { Tree, type ISharedTree, type SharedTree } from "../../shared-tree/index.js";
 import { type JsonCompatibleReadOnly, brand, getOrAddEmptyToMap } from "../../util/index.js";
-import { schematizeFlexTree, treeTestFactory } from "../utils.js";
+import { treeTestFactory } from "../utils.js";
+import {
+	SchemaFactory,
+	TreeViewConfiguration,
+	type InsertableTreeNodeFromImplicitAllowedTypes,
+	type ITree,
+	type TreeView,
+} from "../../simple-tree/index.js";
 
 // Notes:
 // 1. Within this file "percentile" is commonly used, and seems to refer to a portion (0 to 1) or some maximum size.
 // While it would be useful and interesting to have some distribution of op sizes and measure some percentile from that distribution,
 // that does not appear to be what these tests are doing.
-// 2. Data from these tests are just printed: no other data collection is done. If a comparison is desired, manually run the tests before and after.
-// 3. Major changes in these sizes (regressions, optimizations or the tests not collecting what they should) do not make these tests fail.
-// 4. These tests are currently implemented as integration tests, meaning they use lots of dependencies and high level APIs.
-// They could be reimplemented targeted the lower level APIs if desired.
-// 5. "large" node just get a long repeated string value, not a complex tree, so tree encoding is not really covered here.
+// 2. Major changes in these sizes (regressions, optimizations or the tests not collecting what they should) do not make these tests fail.
+// 3. These tests are currently implemented as integration tests, meaning they use lots of dependencies and high level APIs.
+// They could be reimplemented targeted the lower level APIs if desired (just call the op encoding functions)
+// 4. "large" node just get a long repeated string value, not a complex tree, so tree encoding is not really covered here.
 // TODO: fix above issues.
 
-const builder = new SchemaBuilder({ scope: "opSize" });
-
-const childSchema = builder.object("Test:Opsize-Bench-Child", {
-	data: leaf.string,
-});
-const parentSchema = builder.object("Test:Opsize-Bench-Root", {
-	children: builder.sequence(childSchema),
-});
-
-const fullSchemaData = builder.intoSchema(parentSchema);
+const builder = new SchemaFactory("opSize");
```

Comment:
nit: I feel like we should change the name to factory or something similar just so we don't mix the terminology